osu.Game.Rulesets.Mania/Skinning/ManiaLegacySkinTransformer.cs

```diff
@@ -47,15 +78,15 @@ private void sourceChanged()
 
         public Drawable GetDrawableComponent(ISkinComponent component)
         {
+            if (!isLegacySkin.Value || !hasKeyTexture.Value)
```

Comment:
are you sure this hasn't changed behaviour? since it's now applying to other cases it wasn't before.